qcodes/utils/wrappers.py

```diff
@@ -320,39 +329,37 @@ def save_device_image(sweeptparameters):
                             '{:03d}'.format(counter)), title)
 
 
-def do1d(inst_set, start, stop, num_points, delay, *inst_meas, do_plots=True):
+def _do_measurement(loop: Loop, set_params: tuple, meas_params: tuple,
+                    do_plots: Optional[bool]=True) -> Tuple[QtPlot, DataSet]:
     """
+    The function to handle all the auxiliary magic of the T10 users, e.g.
+    their plotting specifications, the device image annotation etc.
+
+    If ax3 is None, ax2 is considered the data axis (non-setpoint).
 
     Args:
-        inst_set:  Instrument to sweep over
-        start:  Start of sweep
-        stop:  End of sweep
-        num_points:  Number of steps to perform
-        delay:  Delay at every step
-        *inst_meas:  any number of instrument to measure and/or tasks to
-          perform at each step of the sweep
-        do_plots: Default True: If False no plots are produced.
-            Data is still saved
-             and can be displayed with show_num.
+        loop: The QCoDeS loop object describing the actual measurement
+        set_params: tuple of tuples. Each tuple is of the form
+            (param, start, stop)
+        meas_params: tuple of parameters to measure
 
     Returns:
-        plot, data : returns the plot and the dataset
-
+        (plot, data)
     """
+    parameters = [sp[0] for sp in set_params] + list(meas_params)
+    _flush_buffers(*parameters)
 
-    # try to flush VISA buffers at the beginning of a measurement
-    _flush_buffers(inst_set, *inst_meas)
-
-    # make a variable for auto pre-scaling the plot
-    startranges = {inst_set.label: (start, stop)}
+    # startranges for _plot_setup
+    startranges = dict(zip((sp[0].label for sp in set_params),
+                           ((sp[1], sp[2]) for sp in set_params)))
 
     interrupted = False
-    loop = qc.Loop(inst_set.sweep(start,
-                                  stop,
-                                  num=num_points), delay).each(*inst_meas)
+
     data = loop.get_data_set()
 
-    plottables = _select_plottables(inst_meas)
+    # this is always the identity operation, is it not?
```

Comment:
Yes, hence the comment below, since we now explicitly ask for the plottables in `meas_param`. We can remove this line and also the `_select_plottables` function altogether.